### ISSUE TITLE
Dynamic TCP port exhaustion

### ISSUE THREAD

**Issue by scptre** [mtconnect/cppagent#136]
We are currently dealing with a problem when connect to a number of MTConnect machines.

We are having a problem where we are running out of available dynamic ports in windows when connecting to 20 or more MTConnect agents from the same server. The currently implementation of the HttpServer which is used in the CppAgent is from a C++ library called dlib. The currently HttpServer implementation in the dlib library only supports http 1.0, this version of http used a keep-alive header to specify that the port was to be left open and reused for addition requests but this is not coded into the dlib implementation. Because of this each request to the agent for data causes a new port to be opened. The default number of dynamic ports on a windows install is 4000. When a port is closed the OS holds it in a TIMED-WAIT state in case delayed packets are received and remains in this state for 5 minutes. With 20 agent connections being polled every 1sec you can get 6000 ports waiting in TIMED_WAIT state causing port exhaustion.

A new http server implementation needs to be adopted by the CppAgent to fix this issue unless they are going to implement the necessary changes in the dlib http server implementation to fix this issue.

I know this is not an issue with how we are handing ports as we are using the HttpClient in DotNet to connect to the agent and I can recreate the port exhaustion by hitting the /current on the agent with chrome over and over again.

**Comment by wsobel** [comment]
The agent is only supporting HTTP 1.0 right now and does not support Keep-Alive. With the current HTTP server implementation, it is not possible to support this. We do not specify the Keep-Alive header: 

```
HTTP/1.0 200 OK
Content-Length: 6936
Content-Type: text/xml

````

The solution We are planning to switch to boost and use beast which will address this issue. Any help migrating and refactoring the code would be greatly appreciated--especially if you would like this faster.

Windows 10 a limit of 16,384. 4k was a pre-vista limit.

**Comment by wsobel** [comment]
Have you thought about streaming data using `interval` with the sample to continuously push deltas? Also works with current.

**Comment by wsobel** [comment]
This will be addressed in the boost version slated for 1.8.
